### PR TITLE
Change Accounts and Oauth to use Async methods.

### PR DESCRIPTION
**Pull request by edimarlnx**
Change Accounts and Oauth to use Async methods.

[Work in Progress]

In this PR, we are starting changes to remove dependencies of Fibers in packages related to Accounts and OAuth. 

In this phase, we will change all codes that depend on Fibers to work synchronously to async methods. 
For example: When waiting for HTTP requests, methods with Meteor.wrapAsync, etc.

Change Oauth packages
- [x] Google
- [x] Github
- [x] Facebook
- [x] Meteor developer
- [x] Password
- [x] Twitter
- [x] Meetup
- [x] Weibo

Test accounts packages
- [x] Google
- [x] Github
- [x] Facebook
- [x] Meteor developer
- [x] Password
- [x] Twitter
- [x] Meetup 
- [x] Weibo 

_* The OAuth packages Meetup and Weibo didn't test because we don't have service accounts for these services._

### CODE REVIEW

**Review by radekmie — COMMENTED** [review]
It's important to document that a few accessible functions (e.g., `Accounts._checkPassword`) are now `async`. I use it in my projects, and I don't think that's uncommon.

Similarly, a guide on how to migrate custom `loginWithX` packages is crucial here (even if it'd say "it'll just work").

**Comment by zodern** [comment]
I think https://github.com/meteor/meteor/issues/10672 will need to be fixed since there will probably be problems if some of these methods are not blocking, especially the login and logout methods.

**Comment by edimarlnx** [comment]
> It's important to document that a few accessible functions (e.g., `Accounts._checkPassword`) are now `async`. I use it in my projects, and I don't think that's uncommon.
I changed it to preserve compatibility mode. 


> Similarly, a guide on how to migrate custom `loginWithX` packages is crucial here (even if it'd say "it'll just work").
I will see it.

Thank you!

File: packages/facebook-oauth/facebook_server.js [hunk 1]
```diff
@@ -1,16 +1,16 @@
 Facebook = {};
 import crypto from 'crypto';
-import { Accounts } from 'meteor/accounts-base';
+import {Accounts} from 'meteor/accounts-base';
```
Comment (StorytellerCZ) [inline packages/facebook-oauth/facebook_server.js:3]: This seems to be some linting happening here. I don't think this is the correct one.

**Review by StorytellerCZ — CHANGES_REQUESTED** [review]

**Comment by perbergland** [comment]
There are a bunch of sync mongodb updates left in accounts_server.js (`users.update(...)`), esp for the token handling method like _insertHashedLoginToken (an example of a method we use in our code to implement SSO login).

All these methods need to become async too and use the mongo async methods, preferrably using the established pattern of adding an async version for now. Is that also planned for 2.9 or is that work saved for a 2.9.x release?

File: packages/google-oauth/google_server.js [hunk 2]
```diff
@@ -56,98 +62,113 @@ const getServiceDataFromTokens = tokens => {
...
+  return returnValue;
 };
 
-Accounts.registerLoginHandler(request => {
+Accounts.registerLoginHandler(async (request) => {
   if (request.googleSignIn !== true) {
     return;
   }
-
+  console.log({ request });
```
Comment (Grubba27) [inline packages/google-oauth/google_server.js:83]: Shouldn't this be removed?

File: packages/google-oauth/google_server.js [hunk 3]
```diff
@@ -56,98 +62,113 @@ const getServiceDataFromTokens = tokens => {
...
-
-  return Accounts.updateOrCreateUserFromExternalService("google", {
-    id: request.userId,
-    idToken: request.idToken,
-    accessToken: request.accessToken,
-    email: request.email,
-    picture: request.imageUrl,
-    ...result.serviceData,
-  }, result.options);
+  console.log({ result });
```
Comment (Grubba27) [inline packages/google-oauth/google_server.js:110]: And this one

**Pull request merged**